### ISSUE TITLE
Overplotting a MOC on a custom WCS projected figure with MOC.fill()

### ISSUE THREAD

**Issue by AlanLoh** [cds-astro/mocpy#69]
Hello,

I was trying to add MOC overlays to my all-sky plots using the `MOC.fill()` method but the resulting graph seems to have its axes modified.

I can try to give a minimal example, using a custom WCS object to define a matplotlib subplot projection.
Then, I create a `MOC` object and overplot it using `moc.fill(ax=ax, wcs=wcs)`.

```python
import matplotlib.pyplot as plt
from astropy.wcs import WCS
import numpy as np
import healpy as hp
from mocpy import MOC
import astropy.units as u

# Generate an abritrary WCS object
wcs = WCS(naxis=2)
wcs.wcs.ctype = ["RA---AIT", "DEC--AIT"] 
wcs.wcs.crval = [0.0, 0.0]
wcs.wcs.cdelt = np.array([-0.675, 0.675])
wcs.wcs.crpix = [240.5, 120.5]

# Get HEALPix cell centers
nside = 64
ra, dec = hp.pix2ang(
    nside,
    np.arange(hp.nside2npix(nside)),
    nest=False,
    lonlat=True
)

# MPL figure with WCS projection
fig = plt.figure(figsize=(15, 7))
ax = plt.subplot(
    projection=wcs
)

# Plot the HEALPix centers (just to test)
ax.scatter(ra, dec, 0.1, transform=ax.get_transform("world"))
ax.grid()

# Define a MOC union
moc1 = MOC.from_cone(
    lon=60*u.deg,
    lat=60*u.deg,
    radius=20*u.deg,
    max_depth=10,
    delta_depth=2
)
moc2 = MOC.from_cone(
    lon=0*u.deg,
    lat=0*u.deg,
    radius=10*u.deg,
    max_depth=10,
    delta_depth=2
)
moc = moc1.union(moc2)

# Plot the MOC in the current figure
moc.fill(ax=ax, wcs=wcs)
```

This is what I get:

![moc_bad_proj](https://user-images.githubusercontent.com/24960512/144453668-677cae66-90c1-43f8-a9b5-231685aa8cb4.png)

The inner plot seems right, however the `x_lim` and `y_lim` are modified by `mocpy/mocpy/moc/plot/axis_viewport.py`.

If I comment:
 https://github.com/cds-astro/mocpy/blob/48dbeef6a182a58dd178a1b5dbdd1527a40e2992/mocpy/moc/plot/fill.py#L86 
I get what I expect:
![moc_good_proj](https://user-images.githubusercontent.com/24960512/144453701-403433c5-e34a-4e69-8740-597c72343e09.png)

Am I doing something wrong?
If not, would it be possible to add an option to the plotting methods to deactivate the call to `axis_viewport` please?

Many thanks,

(Tested on Python 3.7, mocpy 0.10.0)

**Comment by bmatthieu3** [comment]
Hi @AlanLoh 

From the online doc example ([here](https://cds-astro.github.io/mocpy/examples/examples.html#loading-and-plotting-the-moc-of-sdss)) there is a special class inheriting from the astropy WCS class and called World2ScreenMPL.
You should try it. This is a good idea to directly make it work by passing a pure astropy WCS like you intuitively did but this is as it is at the moment. Thus, we should invest time to see if commenting the line does not break any other examples. You can open a new PR with the modif if you want to continue the discussion on that.

On the other hand World2ScreenMPL has a nice constructor where you can specify the fov in astropy.angle you want your view to fit in.

Here is the changes, it works for me. Tell me if you have a problem running it.
```python
import matplotlib.pyplot as plt
import numpy as np
import healpy as hp
from mocpy import MOC, World2ScreenMPL
import astropy.units as u
from astropy.coordinates import Angle, SkyCoord

# Get HEALPix cell centers
nside = 64
ra, dec = hp.pix2ang(
    nside,
    np.arange(hp.nside2npix(nside)),
    nest=False,
    lonlat=True
)

# MPL figure with WCS projection
fig = plt.figure(figsize=(15, 7))

# Define a MOC union
moc1 = MOC.from_cone(
    lon=60*u.deg,
    lat=60*u.deg,
    radius=20*u.deg,
    max_depth=10,
    delta_depth=2
)
moc2 = MOC.from_cone(
    lon=0*u.deg,
    lat=0*u.deg,
    radius=10*u.deg,
    max_depth=10,
    delta_depth=2
)
moc = moc1.union(moc2)

with World2ScreenMPL(fig, 
        fov=360 * u.deg,
        center=SkyCoord(0, 20, unit='deg', frame='icrs'),
        coordsys="icrs",
        rotation=Angle(0, u.degree),
        projection="AIT") as wcs:
    ax = fig.add_subplot(1, 1, 1, projection=wcs)

    # Plot the HEALPix centers (just to test)
    ax.scatter(ra, dec, 0.1, transform=ax.get_transform("world"))
    ax.grid()

    # Plot the MOC in the current figure
    moc.fill(ax=ax, wcs=wcs)
```

Thanks,

**Comment by AlanLoh** [comment]
Thanks @bmatthieu3 for your quick reply.

Indeed, I used to plot MOCs using `World2ScreenMPL`.
However, for some part of my code, I have to define a pure WCS object, and this is why I have tried to see if it would be suitable for plotting MOC objects.

Yes to be honest, commenting this line was the quickest fix I could find to solve this (and then it works like a charm). But on the other hand I don't know if anything else may break.

But you are right, I could stick with `World2ScreenMPL`, no worries.

Merci encore,
À bientôt
